r/tests/testthat/test-dataset.R

```diff
@@ -1502,3 +1502,12 @@ test_that("Collecting zero columns from a dataset doesn't return entire dataset"
     c(32, 0)
   )
 })
+
+# see https://issues.apache.org/jira/browse/ARROW-12791
+test_that("Error if no format specified and files are not parquet", {
+  skip_if_not_available("parquet")
+  expect_error(
+    open_dataset(csv_dir, partitioning = "part"),
+    regexp = "Did you mean to specify a 'format'? other than the default (parquet)?"
+  )
+})
```

Comment:
Technically, for completeness, we should test that the extra error message *isn't* shown if you explicitly provided `format = "parquet"`. Maybe this would work?

```suggestion
  expect_failure(
    expect_error(
      open_dataset(csv_dir, partitioning = "part", format = "parquet"),
      "Did you mean to specify a 'format'"
    )
  )
})
```